hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsDatasetImpl.java

```diff
@@ -1050,8 +1050,10 @@ static File moveBlockFiles(Block b, ReplicaInfo replicaInfo, File destdir)
       File dstFile)
       throws IOException {
     // Create parent folder if not exists.
-    srcReplica.getFileIoProvider()
+    boolean isDirCreated = srcReplica.getFileIoProvider()
         .mkdirs(srcReplica.getVolume(), dstFile.getParentFile());
+    LOG.trace("Dir creation of {} on volume {} {}", dstFile.getParentFile(),
```

Comment:
File.getParentFile() allocates and returns a new string. It's not the kind of parameter you want to log only at TRACE level.

I doubt it'd have a meaningful performance penalty though because the actual file system hard link probably spend much more time.

Reply:
Updated to use already extracted objects so that only if TRACE level is enabled, those objects' toString() will be extracted. Without TRACE level being enabled, placeholder object's toString() won't be called so perf penalty (if applicable) will be only restricted to TRACE level logging. Does this sound good?